File: test/octave/mRBDL/Tools/PlotCoMInertia.m
function PlotCoMInertia(model, q)
%PlotCoMInertia - Plot CoM point and Inertia on model
%
% Syntax: PlotCoMInertia(model)
%
% Long description
    % Plot CoM 
    hold on;
    pos_com = [];
    num = max(size(model.idcomplot,1), size(model.idcomplot,2));
    for i=1:num
        pos_com(:,i) = CalcBodyToBaseCoordinates(model, q, model.idcomplot(i), model.CoM{i});
        plot3(pos_com(1, i),pos_com(2, i),pos_com(3, i), '*');
    end

    % Plot Inertia
    for i=1:num
        lxyz(:,i) = CalcInertiaCubiod(...
        [model.Inertia{i}(1,1), model.Inertia{i}(2,2), model.Inertia{i}(3,3)]',...
         model.Mass{i});
        PlotInertiaCuboid(lxyz(:,i), pos_com(:,i));
    end
    axis([-0.3 -0.3+0.8 -0.15 -0.15+0.8 -0.1 -0.1+0.8]);
end

File: src/jbdl/rbdl/tools/plot_com_inertia.py
import numpy as np
from mpl_toolkits.mplot3d import Axes3D
from jbdl.rbdl.kinematics import calc_body_to_base_coordinates
from jbdl.rbdl.tools import clac_inertia_cuboid
from jbdl.rbdl.tools.plot_inertia_cuboid import plot_inertia_cuboid


def plot_com_inertia(model: dict, q: np.ndarray, ax: Axes3D):

    idcomplot = model["idcomplot"]
    com = model["CoM"]
    inertia = model["Inertia"]
    mass = model["Mass"]

    num = len(idcomplot)
    pos_com = []
    for i in range(num):
        pos_com.append(calc_body_to_base_coordinates(model, q, idcomplot[i], com[i]))
        
    pos_com = np.asfarray(np.concatenate(pos_com, axis=1))
    ax.scatter(pos_com[0, :], pos_com[1, :], pos_com[2, :], marker="*")

    for i in range(num):
        lxyz  = clac_inertia_cuboid(np.diag(inertia[i]), mass[i])
        plot_inertia_cuboid(pos_com[:, i:i+1], lxyz, ax)

    return ax
